sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2696,16 +2687,6 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
         sql("SELECT struct(1 a) EXCEPT (SELECT struct(2 A))")
       }.message
       assert(m2.contains("Except can only be performed on tables with the compatible column types"))
-
-      withTable("t", "S") {
-        sql("CREATE TABLE t(c struct<f:int>) USING parquet")
-        sql("CREATE TABLE S(C struct<F:int>) USING parquet")
-        checkAnswer(sql("SELECT * FROM t, S WHERE t.c.f = S.C.F"), Seq.empty)
-        val m = intercept[AnalysisException] {
-          sql("SELECT * FROM t, S WHERE c = C")
-        }.message
-        assert(m.contains("cannot resolve '(t.`c` = S.`C`)' due to data type mismatch"))
```

Comment:
Thank you for pinging me, @cloud-fan .

Since this removal is a real behavior change instead of new test coverage of `comparator.sql`, could you add a  documentation for this?